python/GafferUI/RampPlugValueWidget.py

```diff
@@ -79,7 +79,9 @@ def __init__( self, plug, **kw ) :
 			self.__splineWidget._qtWidget().setMinimumHeight( 50 )
 
 			self.__slider = GafferUI.Slider()
-			self.__slider.setSizeEditable( True )
+			self.__slider.setSizeEditable( not ( plug.getInput() or
```

Comment:
This needs to go in `_updateFromPlug()` so that we respond to any changes in read-only-ness that occur while the UI is open. If you open the UI, then lock the plug, currently you can add points but not move them.